optimisation/ScanSelectionsTree.py

```diff
@@ -200,54 +252,100 @@
+        outDirPlotsPt[iPt].cd(f'DCA_pT{ptMin}-{ptMax}')
+        hDCA_Prompt.SetLineColor(kRed)
+        hDCA_Prompt.Write()
+        hDCA_FD.SetLineColor(kBlue)
+        hDCA_FD.Write()
+        if inputCfg['DCA']['verbose']:
+            cDCA.cd()
+            hDCA_Prompt.Draw('same')
+            hDCA_FD.Draw('same')
+            legDCA = TLegend(0.6, 0.6, 0.75, 0.75)
+            legDCA.AddEntry(hDCA_Prompt,"Prompt")
+            legDCA.AddEntry(hDCA_FD,"FD")
+            legDCA.Draw('same')
+ 
+ 
     for iSet, cutSet in enumerate(itertools.product(*cutRanges)):
         selToApply = ''
         for iCut, (cut, upperLower, varName) in enumerate(zip(cutSet, upperLowerCuts, varNames)):
             if iCut == 0:
                 selToApply = f'{varName}{upperLower}{cut}'
             else:
                 selToApply += f' & {varName}{upperLower}{cut}'
-
         if (iSet+1) % 100 == 0:
             print(f'Time elapsed to test up to cut set number {iSet+1}: {time.time()-startTime:.2f}s', end='\r')
-
         dfPromptPtSel = dfPromptPt.query(selToApply)
         dfFDPtSel = dfFDPt.query(selToApply)
         dfBkgPtSel = dfBkgPt.query(selToApply)
-
         effPrompt, effPromptUnc = ComputeEfficiency(len(dfPromptPtSel), nTotPrompt,
                                                     np.sqrt(len(dfPromptPtSel)), np.sqrt(nTotPrompt))
         effFD, effFDUnc = ComputeEfficiency(len(dfFDPtSel), nTotFD, np.sqrt(len(dfFDPtSel)), np.sqrt(nTotFD))
-
         effTimesAccPrompt = effPrompt * preselEffPrompt * acc
         effTimesAccFD = effFD * preselEffFD * acc
-
         fPrompt, fFD = GetPromptFDFractionFc(
             effTimesAccPrompt, effTimesAccFD, crossSecPrompt, crossSecFD, RaaPrompt, RaaFD)
-
         hMassBkg = TH1F(f'hMassBkg_pT{ptMin}-{ptMax}_cutSet{iSet}', ';#it{M} (GeV/#it{c});Counts',
                         400, min(dfBkgPtSel['inv_mass']), max(dfBkgPtSel['inv_mass']))
         hMassSignal = TH1F(f'hMassSignal_pT{ptMin}-{ptMax}_cutSet{iSet}', ';#it{M} (GeV/#it{c});Counts', 400, min(
             dfPromptPtSel['inv_mass']), max(dfPromptPtSel['inv_mass']))
+
         fill_hist(hMassBkg, dfBkgPtSel['inv_mass'].values)
         fill_hist(hMassSignal, list(dfPromptPtSel['inv_mass'].values)+list(dfFDPtSel['inv_mass'].values))
 
-        if dfSecPeakPrompt and dfSecPeakFD:
+        # SecPeak
+        if dfSecPeakPrompt and dfSecPeakFD: 
             hMassSeaPeak = TH1F(f'hMassSignal_pT{ptMin}-{ptMax}_cutSet{iSet}', ';#it{M} (GeV/#it{c});Counts', 400, min(
                 dfSecPeakPrompt['inv_mass']), max(dfSecPeakPrompt['inv_mass']))
             fill_hist(hMassSeaPeak, list(dfSecPeakPrompt['inv_mass'].values)+list(dfSecPeakFD['inv_mass'].values))
         else:
             hMassSeaPeak = None
-
         # expected signal
         expSignal = GetExpectedSignal(crossSecPrompt, ptMax-ptMin, 1., effTimesAccPrompt,
-                                      fPrompt[0], 1., 1., nExpEv, sigmaMB, Taa, RaaPrompt)
```

Comment:
Be careful that RaaPrompt and RaaFD are spline objects in case of Raa read from txt file